Automattic/wp-calypso
# Home: Fix unhandled exception on layout query

Fixes https://github.com/Automattic/dotcom-forge/issues/3601
## Proposed Changes

This PR fixes the issue mentioned at p1693373541277569-slack-C04U5A26MJB.

### Why this error occurred?
Inside the `Home` component we use a hook to fetch the home layout ( `useHomeLayoutQuery` ).
If the request fails, we end up having a `layout` property that is `null`, the layout components are trying to load and we try to access the properties of an object that doesn't exist.

## Testing Instructions

### Test 1: Functionality
1. Apply the following diff ( not on this branch ).
```diff --git a/client/data/home/use-home-layout-query.ts b/client/data/home/use-home-layout-query.ts
index f089d41a11..f02d329e7d 100644
--- a/client/data/home/use-home-layout-query.ts
+++ b/client/data/home/use-home-layout-query.ts
@@ -29,6 +29,9 @@ export function fetchHomeLayout(
 	siteId: number | null,
 	query: HomeLayoutQueryParams = {}
 ): Promise< unknown > {
+	return new Promise( ( resolve, reject ) => {
+		reject( new Error( 'Emulating an error....' ) );
+	} );
 	return wpcom.req.get(
 		{
 			path: `/sites/${ siteId }/home/layout`,
```

2. Visit `/home`.
3. Open the dev console and wait for a couple of seconds.
4. Ensure that you see the following error and the page doesn't load:
![emu_error](https://github.com/Automattic/wp-calypso/assets/497103/282c9563-deb5-417b-a08a-b9f17e4007b2)
5. Checkout this PR, keeping the applied diff.
6. Visit `/home`
7. Ensure that the page is not crashing.

### Test 2: Analytics
1. Checkout this PR, keeping the applied diff.
2. Enable debugging to be able to see analytic events.
3. Visit `/home`
4. Ensure that the page is not crashing.
5. Ensure that you see `calypso_customer_home_my_site_view_layout_error` being recorded.


## Pre-merge Checklist
